src/Runners/PHPUnit/WrapperRunner.php

```diff
@@ -61,10 +64,16 @@ private function assignAllPendingTests(): void
                 }
 
                 $this->flushWorker($worker);
+                if ($batchSize && $this->batches[$token] > $batchSize) {
```

Comment:
```suggestion
                if (null !== $batchSize && $this->batches[$token] === $batchSize) {
```
If you use the `>` operator you are actually executing `maxBatchSize + 1` tests in each batch.
`=== ` is needed